### ISSUE TITLE
aws eks update-cluster-config throwing InvalidParameterException

### ISSUE THREAD

**Issue by gandhiamarnadh** [aws/aws-cli#4072]
trying updateupdate-cluster-config command to update subnets and it throws
`An error occurred (InvalidParameterException) when calling the UpdateClusterConfig operation: Invalid type for cluster update.`

**Comment by justnance** [comment]
@gandhiamarnadh - Thank you for your post.  This error is usually generated from the service and not the CLI.  The  [updateupdate-cluster-config command](https://docs.aws.amazon.com/cli/latest/reference/eks/update-cluster-config.html) is used to enable or disable public and private access to your cluster's Kubernetes API server endpoint.  It is also used to enable or disable exporting the Kubernetes control plane logs for your cluster to CloudWatch Logs.

Please provide an example of the parameter values used that returned the error so we can try to reproduce the problem.

**Comment by steinbachr** [comment]
I'm having the same error as @gandhiamarnadh. See below for command I'm attempting to run (following directions from https://docs.aws.amazon.com/cli/latest/reference/eks/update-cluster-config.html):
```
aws eks update-cluster-config --name eks-staging --resources-vpc-config subnetIds=subnet-0679a7d73e54b4e8b,subnet-0baceb7e7b5d8c129,subnet-0e8579fa2f66799dc,securityGroupIds=sg-043c39c25a7caeb52,endpointPublicAccess=true,endpointPrivateAccess=false
```

Response:
```
An error occurred (InvalidParameterException) when calling the UpdateClusterConfig operation: Invalid type for cluster update.
```

CLI version info:
```
[15:27] ➜  mm-cac-charts git:(master) ✗ aws --version
aws-cli/1.16.141 Python/3.7.3 Darwin/18.2.0 botocore/1.12.131
```

**Comment by justnance** [comment]
@gandhiamarnadh and @steinbachr - Thank you for your feedback.  It was helpful.  I was able to reproduce the issue and found the CLI is `securityGroupIds` and `subnetIds` but EKS service is returning the "Invalid type for cluster update" error.  This was tested with current and correct Security Group IDs and Subnet IDS.  It was also tested with incorrect IDs and the service returned the same error.  

If `securityGroupIds` and `subnetIds` are excluded from the command, the behavior works as expected.  I also found if the endpointPublicAccess/endpointPrivateAccess is sent with the current values where no change is sent the service returns:  "Cluster is already at the desired configuration with endpointPrivateAccess: false and endpointPublicAccess: true".  This is expected behavior and working correctly.  

I also found the [CLI documentation](https://docs.aws.amazon.com/cli/latest/reference/eks/update-cluster-config.html) and the [API documentation](https://docs.aws.amazon.com/eks/latest/APIReference/API_UpdateClusterConfig.html#AmazonEKS-UpdateClusterConfig-request-resourcesVpcConfig) needs review and improvements. reviewed our online documentation finding it needs improvements.  

I have escalated my findings internally to the EKS service team and documentation writers.  Since the CLI does not control this behavior, it would be better for this issue to be posted on the EKS forums for follow up with the service team. 

Thanks again for reporting this issue.
